code/modules/reagents/chemistry/items.dm

```diff
@@ -0,0 +1,163 @@
+///if the pHmeter gives a detailed output
+#define DETAILED_CHEM_OUTPUT 1 
+///if the pH meter gives a shorter output
+#define SHORTENED_CHEM_OUTPUT 0 
+
+/*
+* a pH booklet that contains pH paper pages that will change color depending on the pH of the reagents datum it's attacked onto
+*/
+/obj/item/pHbooklet
+	name = "pH indicator booklet"
+	desc = "A booklet containing paper soaked in universal indicator."
+	icon_state = "pHbooklet"
+	icon = 'icons/obj/chemical.dmi'
+	item_flags = NOBLUDGEON
+	resistance_flags = FLAMMABLE
+	w_class = WEIGHT_CLASS_TINY
+	///How many pages the booklet holds
+	var/number_of_pages = 50
+
+//A little janky with pockets
+/obj/item/pHbooklet/attack_hand(mob/user, act_intent = user.a_intent, unarmed_attack_flags)
+	if(user.get_held_index_of_item(src))//Does this check pockets too..?
+		if(number_of_pages == 50)
+			icon_state = "pHbooklet_open"
+		if(!number_of_pages)
+			to_chat(user, "<span class='warning'>[src] is empty!</span>")
+			add_fingerprint(user)
+			return
+		var/obj/item/pHpaper/page = new(get_turf(user))
+		page.add_fingerprint(user)
+		user.put_in_active_hand(page)
```

Comment:
```suggestion
		var/obj/item/pHpaper/page = new(get_turf(src))
```
At the end of this proc the user will try to grab the newly created page, already.
Also, better to spawn it in the booklet turf instead, no?

Reply:
Is there not a return statement at the end of that nest if? I think the two differences means that you get different effects if it's in a pocket or not. The idea is that if you click on it in your other hand, your active hand gets a page, whereas if you click on it in a pocket, you move the item. Additionally, dragging it to yourself will give you a page too. I'll test it quickly and get back to you - I should probably comment this more anyways.

Reply:
Removing those lines drops the paper to the floor instead. 
```c
var/I = user.get_active_held_item()
if(!I)
    user.put_in_active_hand(src)
```
This part is about allowing the item to move around in pockets, since it moves the booklet not the paper.

Reply:
You're right. 👍